openarm/damiao/encoding.py

```diff
@@ -0,0 +1,752 @@
+_READ_REGISTER_CODE = 0x33
+_STATE_CODE = 0x11
+
+
+@dataclass
+class MotorLimits:
+    """Motor physical limits for parameter scaling.
+
+    Reference: DM_CAN.py Limit_Param array structure
+    """
+
+    q_max: float  # Maximum position in radians
+    dq_max: float  # Maximum velocity in radians/second
+    tau_max: float  # Maximum torque in Nm
+
+
+@dataclass
+class MitControlParams:
+    """MIT control parameters for Damiao motor control.
+
+    Reference: DM_CAN.py controlMIT function parameters lines 90-99
+    """
+
+    kp: float  # Proportional gain (0-500)
+    kd: float  # Derivative gain (0-5)
+    q: float  # Desired position in radians
+    dq: float  # Desired velocity in radians/second
+    tau: float  # Desired torque in Nm
+
+
+T = TypeVar("T", int, float)  # Type variable for register value types
+
+
+@dataclass
+class RegisterResponse(Generic[T]):
+    """Register operation response data from Damiao motor.
+
+    Reference: DM_CAN.py __process_set_param_packet function lines 291-315
+    """
+
+    motor_id: int  # Motor slave ID that responded
+    register: int  # Register address that was accessed
+    value: T  # Register value (int for uint32, float for float32)
+    command: int  # Command code (0x55 for write, 0x33 for read)
+
+
+@dataclass
+class AckResponse:
+    """Simple acknowledgment response data from Damiao motor.
+
+    Reference: DM_CAN.py save_motor_param and other simple operations
+    """
+
+    motor_id: int  # Motor slave ID that responded
+    command: int  # Command code that was executed
+    success: bool  # Whether the operation was successful
+
+
+@dataclass
+class MotorState:
+    """Motor state response data from Damiao motor.
+
+    Reference: DM_CAN.py MotorState structure and recv_data function
+    """
+
+    position: float  # Motor position in radians
+    velocity: float  # Motor velocity in radians/second
+    torque: float  # Motor torque in Nm
+    temp_mos: int  # MOS temperature
+    temp_rotor: int  # Rotor temperature
+
+
+@dataclass
+class PosVelControlParams:
+    """Position and velocity control parameters for Damiao motor control.
+
+    Reference: DM_CAN.py control_Pos_Vel function parameters lines 130-135
+    """
+
+    position: float  # Desired position in radians
+    velocity: float  # Desired velocity in radians/second
+
+
+@dataclass
+class VelControlParams:
+    """Velocity control parameters for Damiao motor control.
+
+    Reference: DM_CAN.py control_Vel function parameters lines 150-153
+    """
+
+    velocity: float  # Desired velocity in radians/second
+
+
+@dataclass
+class PosForceControlParams:
+    """Position and force control parameters for Damiao motor EMIT control.
+
+    Reference: DM_CAN.py control_pos_force function parameters lines 165-170
+    """
+
+    position: float  # Desired position in radians
+    velocity: float  # Desired velocity in radians/second (scaled by 100)
+    current_norm: float  # Normalized current 0-1 (scaled to 0-10000)
```

Comment:
No need to mention about scaling as it's done inside the function, not related to the params:

```suggestion
    velocity: float  # Desired velocity in radians/second 
    current_norm: float  # Normalized current 0-1
```